### PR TITLE
feat: correlation between different versions of cluster-level Addon deployments under different projects.

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:

correlation between different versions of cluster-level Addon deployments under different projects.

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=572766&iterationID=12783&type=TASK)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     correlation between different versions of cluster-level Addon deployments under different projects.        |
| 🇨🇳 中文    |     实现不同项目下，不同版本的集群级别Addon部署的关联         |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/orchestrator/services/addon/addon.go [hunk 1]
```diff
@@ -3072,6 +3149,58 @@ func (a *Addon) deployAddons(req *apistructs.AddonCreateRequest, deploys []dbcli
...
+	}
+	registers, ok := regMap.(*VersionMap)
+	if !ok {
+		return "", "", errors.New("cache data type error")
+	}
+	configs, ok := confMap.(*VersionMap)
+	if !ok {
+		return "", "", errors.New("cache data type error")
+	}
+	for _, r := range *registers {
```
Comment (iutx) [inline internal/tools/orchestrator/services/addon/addon.go:3169]: Code redundancy. Similar logic for registercenter and configcenter.
Assert target all is `VersionMap`.

File: internal/tools/orchestrator/services/addon/addon_handler.go [hunk 2]
```diff
@@ -67,6 +67,19 @@ func (a *Addon) AttachAndCreate(params *apistructs.AddonHandlerCreateItem) (*api
 	return a.addonAttach(addonSpec, addonDice, params)
 }
 
+func (a *Addon) parseAddonDice(addon apistructs.ExtensionVersion) (diceyml.Object, error) {
+	var dice diceyml.Object
+	diceBytes, err := json.Marshal(addon.Dice)
```
Comment (iutx) [inline internal/tools/orchestrator/services/addon/addon_handler.go:72]: Use internal parser.
```
diceyml.New()
```

**Review by iutx — CHANGES_REQUESTED** [review]
@CeerDecy

**Review by iutx — APPROVED** [review]
@sfwn PTAL

File: internal/tools/orchestrator/dbclient/tmc_instance.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dbclient
+
+import (
+	"time"
+
+	"github.com/jinzhu/gorm"
```
Comment (sfwn) [inline internal/tools/orchestrator/dbclient/tmc_instance.go:20]: you'd better use `https://github.com/go-gorm/gorm`

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/orchestrator/dbclient/tmc_instance.go (hunk 3 above)
Comment (CeerDecy) [reply internal/tools/orchestrator/dbclient/tmc_instance.go:20]: no need to change. in DbClient use `jinzhu/gorm` , here only `err` is judged.

**Review by sfwn — APPROVED** [review]

**Pull request merged**